neo/Consensus/ConsensusService.cs

```diff
@@ -165,126 +150,111 @@ private static ulong GetNonce()
+            if (view_number == 0)
+                context.Reset(wallet);
+            else
+                context.ChangeView(view_number);
+            if (context.MyIndex < 0) return;
+            Log($"initialize: height={context.BlockIndex} view={view_number} index={context.MyIndex} role={(context.MyIndex == context.PrimaryIndex ? ConsensusState.Primary : ConsensusState.Backup)}");
+            if (context.MyIndex == context.PrimaryIndex)
             {
-                if (view_number == 0)
-                    context.Reset(wallet);
-                else
-                    context.ChangeView(view_number);
-                if (context.MyIndex < 0) return;
-                Log($"initialize: height={context.BlockIndex} view={view_number} index={context.MyIndex} role={(context.MyIndex == context.PrimaryIndex ? ConsensusState.Primary : ConsensusState.Backup)}");
-                if (context.MyIndex == context.PrimaryIndex)
+                context.State |= ConsensusState.Primary;
+                if (!context.State.HasFlag(ConsensusState.SignatureSent))
                 {
-                    context.State |= ConsensusState.Primary;
-                    if (!context.State.HasFlag(ConsensusState.SignatureSent))
-                    {
-                        FillContext();
-                    }
-                    if (context.TransactionHashes.Length > 1)
-                    {
-                        InvPayload invPayload = InvPayload.Create(InventoryType.TX, context.TransactionHashes.Skip(1).ToArray());
-                        foreach (RemoteNode node in localNode.GetRemoteNodes())
-                            node.EnqueueMessage("inv", invPayload);
-                    }
-                    timer_height = context.BlockIndex;
-                    timer_view = view_number;
-                    TimeSpan span = DateTime.Now - block_received_time;
-                    if (span >= Blockchain.TimePerBlock)
-                        timer.Change(0, Timeout.Infinite);
-                    else
-                        timer.Change(Blockchain.TimePerBlock - span, Timeout.InfiniteTimeSpan);
+                    FillContext();
                 }
-                else
+                if (context.TransactionHashes.Length > 1)
                 {
-                    context.State = ConsensusState.Backup;
-                    timer_height = context.BlockIndex;
-                    timer_view = view_number;
-                    timer.Change(TimeSpan.FromSeconds(Blockchain.SecondsPerBlock << (view_number + 1)), Timeout.InfiniteTimeSpan);
+                    foreach (InvPayload payload in InvPayload.CreateGroup(InventoryType.TX, context.TransactionHashes.Skip(1).ToArray()))
+                        Context.Parent.Tell(new LocalNode.Broadcast
+                        {
+                            Message = Message.Create("inv", payload)
+                        });
                 }
+                TimeSpan span = DateTime.Now - block_received_time;
```

Comment:
Erik, let's consider a startup time here.

`TimeSpan span = DateTime.Now + averageStartupTimeTimeOfKPreviousBlocks - block_received_time;`

But, perhaps, not now, right? It would make harder to detect any different in this refactoring.

Reply:
DateTime.UtcNow please

Reply:
What is the difference, Sharrrgon brothers?

Reply:
The important thing here would be to have something unified in terms of clock and time syncing. The template has almost not so much effects.